test/Log/LoggerFactoryHelperTrait.php

```diff
@@ -19,52 +20,84 @@ trait LoggerFactoryHelperTrait
+    ): ContainerInterface {
+        $methodMap = ArrayUtils::merge($methodMap, $this->registerConfigService($config));
+        $methodMap = ArrayUtils::merge($methodMap, [
+            'has' => [LoggerInterface::class => true],
+            'get' => [LoggerInterface::class => $this->logger],
         ]);
-        $this->container->get('my_logger')->willReturn($this->logger);
 
-        return $this->container->reveal();
+        return $this->mockContainer($methodMap);
     }
 
-    private function createContainerMockWithConfigAndPsrLogger(?array $config = null): ContainerInterface
-    {
-        $this->registerConfigService($config);
-        $this->container->has(LoggerInterface::class)->willReturn(true);
-        $this->container->get(LoggerInterface::class)->shouldBeCalled()->willReturn($this->logger);
+    private function createContainerMockWithConfigAndNotPsrLogger(
+        array $methodMap = [],
+        ?array $config = null
+    ): ContainerInterface {
+        $methodMap = ArrayUtils::merge($methodMap, $this->registerConfigService($config));
+        $methodMap = ArrayUtils::merge($methodMap, [
+            'has' => [LoggerInterface::class => false],
+        ]);
 
-        return $this->container->reveal();
+        return $this->mockContainer($methodMap);
     }
 
-    private function createContainerMockWithConfigAndNotPsrLogger(?array $config = null): ContainerInterface
+    private function registerConfigService(?array $config = null): array
     {
-        $this->registerConfigService($config);
-        $this->container->has(LoggerInterface::class)->willReturn(false);
-        $this->container->get(LoggerInterface::class)->shouldNotBeCalled();
+        $spec = [
+            'has' => [
+                'config' => $config !== null,
+            ],
+        ];
+
+        if ($config !== null) {
+            $spec['get']['config'] = $config;
+        }
 
-        return $this->container->reveal();
+        return $spec;
     }
 
-    private function registerConfigService(?array $config = null): void
+    private function mockContainer(array $methodMap): ContainerInterface
     {
-        $hasConfig            = $config !== null;
-        $shouldBeCalledMethod = $hasConfig ? 'shouldBeCalled' : 'shouldNotBeCalled';
+        $methodMap = ArrayUtils::merge($methodMap, [
+            'has' => [SwooleLoggerFactory::SWOOLE_LOGGER => false],
+        ]);
+        $container = $this->createMock(ContainerInterface::class);
+        foreach ($methodMap as $method => $serviceMap) {
+            $valueMap = [];
+
+            foreach ($serviceMap as $service => $value) {
+                $valueMap[] = [$service, $value];
+            }
+
+            $container->method($method)->will($this->returnValueMap($valueMap));
+        }
 
-        $this->container->has('config')->willReturn($hasConfig);
-        $this->container->get('config')->{$shouldBeCalledMethod}()->willReturn($config);
+        return $container;
     }
```

Comment:
This bit was necessary because PHPUnit mocks are such that the last assertion defined for a method wins. As such, you cannot programmatically add rules for a method via successive calls; instead, you need to define them all at once.

This approach accomplishes that by slinging around a specification of methods + service name/value pairs, setting up a value map in the mock instance based on them. This ensures that if a call is made using different arguments, it will raise an error; otherwise, the call goes through, and the appropriate value is returned.